### DISCUSSION TITLE
i18 pseudolocalization support

### DISCUSSION THREAD

**Discussion by kcubero27** [vercel/next.js#38124] [Help] [answered]
Hi everyone 👋 

I would like to know if Next.js supports pseudolocalization such as https://www.npmjs.com/package/i18next-pseudo.
I've seen some examples but all of them need to create the translations manually.
Is there any chance that Next.js has already this feature enabled?

Thanks!

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I tried this used `next-i18next`, which is listed as `i18next`'s framework for Next.js

1 .Create a `next-18next.config.js`:

```js
module.exports = {
  i18n: {
    defaultLocale: "en",
    locales: ["en", "de"],
    use: [],
    postProcess: "pseudo",
  },
  serializeConfig: false,
};
```

2. Then in `_app`:

```js
import { appWithTranslation, i18n } from "next-i18next";
import Pseudo from "i18next-pseudo";
import nextI18NextConfig from "../next-i18next.config.js";

nextI18NextConfig.i18n.use.push(new Pseudo());

const MyApp = ({ Component, pageProps }) => <Component {...pageProps} />;

export default appWithTranslation(MyApp, nextI18NextConfig);
```

3. And I think that's it, just make sure `next.config.js` has:

```js
const { i18n } = require("./next-i18next.config");

module.exports = {
  i18n,
};
```

I used this https://github.com/isaachinman/next-i18next/tree/master/examples/simple as base, with this package.json:


```json
{
  "name": "v10",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start -p ${PORT:=3000}"
  },
  "dependencies": {
    "i18next-pseudo": "^2.2.1",
    "next": "12.2.0",
    "next-i18next": "11.0.0",
    "react": "18.2.0",
    "react-dom": "18.2.0"
  }
}
```

I removed the `yarn.lock` file and installed fresh.

This was the result:

![Screenshot 2022-06-28 at 23 13 17](https://user-images.githubusercontent.com/21013447/176289035-0e69a959-34fa-46bb-893f-38bb0bd8b211.png)
